src/map/mob.cpp

```diff
@@ -1738,10 +1747,11 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 	if (md->ud.state.force_walk)
 		return false;
 
-	if (DIFF_TICK(tick, md->last_thinktime) < MIN_MOBTHINKTIME)
+	if (DIFF_TICK(tick, md->next_thinktime) < 0)
```

Comment:
Do you really want to trigger this with 0 difference?

Reply:
Yes, if the "end of attack" timer event happens in the same millisecond as they hard AI was executed, we need to run through the AI again because now the attacktimer was cleared which can lead to a different result (usually the mob starting the attack).

I made sure that if we come from lazy -> hard AI that it won't trigger again on the same millisecond in the lazy AI code.